File: src/main/java/com/oshi/libgenericsettings/data/TitleSubtitleExtraSwitchData.java
package com.oshi.libgenericsettings.data;

import android.support.annotation.ColorRes;

/**
 * Title, Subtitle & an extra text. Being use in layout/
 */
public class TitleSubtitleExtraSwitchData extends TitleSubtitleSwitchData {

    protected String extra;
    protected @ColorRes int extraColor;

    public TitleSubtitleExtraSwitchData(String title, String subtitle, String extra) {
        super(title, subtitle);
        this.extra = extra;
    }

    public String getExtra() {
        return extra;
    }

    public void setExtra(String extra) {
        this.extra = extra;
    }

    public int getExtraColor() {
        return extraColor;
    }

    public void setExtraColor(int extraColor) {
        this.extraColor = extraColor;
    }

    @Override
    public int getViewType() {
        return VIEW_TYPE_SETTINGS_TITLE_SUBTITLE_EXTRA_SWITCH;
    }
}

File: src/main/java/com/oshi/libgenericsettings/data/TitleSubtitleExtraSwitchData.kt
package com.oshi.libgenericsettings.data

import android.support.annotation.ColorRes
import com.oshi.libgenericsettings.data.BaseViewTypeData.Companion.SettingsViewTypes

/**
 * Title, Subtitle & an extra text. Being use in layout/view_type_title_subtitle_extra.xml
 */
class TitleSubtitleExtraSwitchData(title : String, subtitle : String, var extra : String) : TitleSubtitleSwitchData(title, subtitle) {

    @ColorRes var extraColor : Int = 0

    override fun getViewType(): SettingsViewTypes {
        return SettingsViewTypes.VIEW_TYPE_SETTINGS_TITLE_SUBTITLE_EXTRA_SWITCH
    }
}